pkg/kubelet/kubelet.go

```diff
@@ -1387,16 +1381,24 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 		return nil
 	}
 
-	// Kill containers associated with unwanted pods and get a list of
-	// unwanted containers that are still running.
-	running, err := kl.killUnwantedPods(desiredPods, runningPods)
+	// Kill containers associated with unwanted pods.
+	err = kl.killUnwantedPods(desiredPods, runningPods)
 	if err != nil {
 		glog.Errorf("Failed killing unwanted containers: %v", err)
+	}
+
+	// Note that we just killed the unwanted pods. This may not have reflected
+	// in the cache. We need to bypass the cach to get the latest set of
+	// running pods to clean up the volumes.
+	// TODO: Evaluate the performance impact of bypassing the runtime cache.
+	runningPods, err = kl.containerManager.GetPods(true)
```

Comment:
Why we pass `true` here? I remember `true` is for getting all pods including dead ones?